From a53216fd3cd1b00d3976b0c4bb03d339256d9317 Mon Sep 17 00:00:00 2001
From: Maxime Chevallier <maxime.chevallier@bootlin.com>
Date: Thu, 6 Dec 2018 15:45:35 +0100
Subject: [PATCH 119/386] phy: cp110-comphy: Initialize both lanes in RXAUI
 mode

RXAUI uses 2 serdes lanes. For now, the consumer only uses a PHY
reference to the PHY representing the first lane. Make wure we
initialize both when using RXAUI.

Change-Id: I383fda8e4350bc0e0ffe3f8e4ce29039fa96d880
Signed-off-by: Maxime Chevallier <maxime.chevallier@bootlin.com>
Reviewed-on: https://sj1git1.cavium.com/6487
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/phy/marvell/phy-mvebu-cp110-comphy.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
index fa6bb87f7829..a956cf764592 100644
--- a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
+++ b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
@@ -258,6 +258,10 @@ static int mvebu_comphy_eth_power_on(struct phy *phy)
 		ret = data->comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
 				 lane->id,
 				 COMPHY_FW_MODE_FORMAT(COMPHY_RXAUI_MODE));
+
+		ret = data->comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
+				 lane->id + 1,
+				 COMPHY_FW_MODE_FORMAT(COMPHY_RXAUI_MODE));
 		break;
 	default:
 		dev_err(priv->dev, "unsupported PHY submode (%d)\n",
-- 
2.17.1

